app/api/orders.py

```diff
@@ -54,6 +54,14 @@ def check_event_user_ticket_holders(order, data, element):
                                      "You cannot update {} of an order".format(element))
 
 
+def is_payment_valid(data, mode):
+    if mode == 'stripe':
+        return data.get('paid_via') and data.get('brand') and data.get('transaction_id') and data.get('exp_month') \
+            and data.get('exp_year') and data.get('last4') and data.get('stripe_token')
+    elif mode == 'paypal':
+        return data.get('paid_via') and data.get('transaction_id')
```

Comment:
How does this guarantee that the payment is actually made at all? Is this data dump_only and can only be set via server, or it can be set via client as well? If it can be set via client, this check is useless

Reply:
Currently these values are set by server only in `helpers/ticketing.py`.
Only paid_via is set by client for now.

Reply:
Currently doesn't matter. Is there any restriction which prevents clients to set these values or they can set it

Reply:
We are not providing any options but as far as I found there are no restrictions too. If they want they can set it

Reply:
Then it is useless. Make these properties dump only, only then these checks work

Reply:
All the properties used are already dump_only props